### ISSUE TITLE
Adding unpublished content to menus

### ISSUE THREAD

**Issue by Graham-72** [backdrop/backdrop-issues#1333]
As admin, I wanted to add an item to an existing menu whilst the item was still unpublished, so that it becomes visible in admin pages though not yet public. It would then become publicly visible as soon as the item (a page) was published by another team member who does not have permission to edit the menu.

What I had to do was briefly publish the draft page so that I could set the menu correctly, then unpublish the page so that the menu item no longer appears. How else could I do this?

**Comment by klonos** [comment]
I see this as a bug actually...

People with the right permissions should be able to add unpublished content to menus. Whether these items are shown or not should be a mater of access permissions (view unpublished content).

**Comment by jenlampton** [comment]
I agree it seems like a bug. It looks like the problem might be the call to `menu_tree_check_access($tree, $node_links);` on line 67 of menu.admin.inc.

I agree we don't want links appearing in the public-facing menu when people don't have access to them, but it seems like admins or those who do have access to edit menus should be able to see **all** the links in the menu.

**Comment by Graham-72** [comment]
>admins or those who do have access to edit menus should be able to see all the links in the menu

:+1:

**Comment by BWPanda** [comment]
This seems like it'd be even more useful/needed now that we can schedule content to be published automatically in the future...
